dvc/scm/git/__init__.py

```diff
@@ -112,31 +124,26 @@ def ignore(self, path, in_curr_dir=False):
         )
         entry, gitignore = self._get_gitignore(path, base_dir)
 
-        ignore_list = []
-        if os.path.exists(gitignore):
-            with open(gitignore, "r") as f:
-                ignore_list = f.readlines()
-            if any(filter(lambda x: x.strip() == entry.strip(), ignore_list)):
-                return
+        ignore_list = Git._process_existing_gitignore(gitignore)
+        if any(filter(lambda x: x.strip() == entry.strip(), ignore_list)):
+            return
 
         msg = "Adding '{}' to '{}'.".format(
             os.path.relpath(path), os.path.relpath(gitignore)
         )
         logger.info(msg)
 
-        self._add_entry_to_gitignore(entry, gitignore, ignore_list)
+        self._add_entry_to_gitignore(entry, gitignore)
 
         self.track_file(os.path.relpath(gitignore))
 
         self.ignored_paths.append(path)
 
     @staticmethod
-    def _add_entry_to_gitignore(entry, gitignore, ignore_list):
-        content = entry
-        if ignore_list:
-            content = "\n" + content
+    def _add_entry_to_gitignore(entry, gitignore):
+
         with open(gitignore, "a", encoding="utf-8") as fobj:
```

Comment:
And here
```suggestion
        with open(gitignore, "a+", encoding="utf-8") as fobj:
            fobj.seek(-1, os.SEEK_END)
            prefix = "" if fobj.read(1) == "\n" else "\n"
            fobj.write("{}{}\n".format(prefix, entry))
```
?

Reply:
So you won't need to write anything when you are checking whether or not entry is already in gitignore above.